src/main/java/reposense/util/TimeUtil.java

```diff
@@ -208,11 +210,27 @@ public static String extractDate(String date) {
 
     /**
      * Parses the given String as a Date based on the {@code CLI_ARGS_DATE_FORMAT}.
-     * Setting setLenient to false prevents unexpected results.
-     * Without it, even with "dd/MM/yyyy HH:mm:ss" format, 11/31/2017 00:00:00 will be parsed to 11/7/2019 00:00:00.
+     * Uses {@code ResolverStyle.STRICT} to avoid unexpected dates like 31/02/2020.
      */
-    public static Date parseDate(String date) throws java.text.ParseException {
-        CLI_ARGS_DATE_FORMAT.setLenient(false);
-        return CLI_ARGS_DATE_FORMAT.parse(date);
+    public static Date parseDate(String date) throws ParseException {
+        try {
+            String[] processDate = date.split("/");
+            if (processDate[0].length() == 1) {
+                // Need to add 0 for single-digit day, like 2/02/2022
+                processDate[0] = '0' + processDate[0];
+            }
+            if (processDate[1].length() == 1) {
+                // Likewise for month
+                processDate[1] = '0' + processDate[1];
+            }
+            date = processDate[0] + "/" + processDate[1] + "/" + processDate[2];
+            return Date.from(ZonedDateTime.parse(date,
+                    CLI_ARGS_DATE_FORMAT.withResolverStyle(ResolverStyle.STRICT)
+                            .withZone(ZoneId.systemDefault())).toInstant());
```

Comment:
I'm not sure if we should revert back to `SimpleDateTime` here. I noticed that I cannot pass in a `zoneId` from the arguments without putting the timezone together with the date.

Reply:
I feel this code is a little too complicated, the given java libraries really should have a way to resolve this.

Reply:
Might want to try reading [this](https://stackoverflow.com/questions/42903391/convert-simpledateformat-to-datetimeformatter)

Reply:
For this part of the code, we only need to store the date (the time will be discarded later on). We don't have to worry about the time zone here, so using system default is fine. The time zone information will be added later in `ArgsParser.java`.